### PR TITLE
Feature/post tokenizer treatement

### PR DESCRIPTION
**Pull request by Mironor**

### CODE REVIEW

File: parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala [hunk 1]
```diff
@@ -27,9 +56,8 @@ object Tokens extends CrfTokenable {
     val inputWithoutCounties = removeCounties(upperInput)
 
     val tokens = inputWithoutCounties
-      .replace(" - ", "-")
-      .replace("- ", "-")
-      .replace(" -", "-")
+      .replaceAll("(\\d+) ?- ?(\\d+)", "$1-$2")
```
Comment (iRhysBradbury) [inline parsers/src/main/scala/uk/gov/ons/addressIndex/parsers/Tokens.scala]: Possible improvement to this regex

`.replaceAll("(\\d+).*?-.*?(\\d+)", "$1-$2")`

File: parsers/src/test/scala/uk/gov/ons/addressIndex/parsers/postTokenizeTreatmentTest.scala [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    )
+
+    // When
+    val actual = Tokens.postTokenizeTreatment(input)
+
+    // Then
+    actual shouldBe expected
+  }
+
+  it should "NOT transform buildingName if it doesn't follow the pattern" in {
```
Comment (iRhysBradbury) [inline parsers/src/test/scala/uk/gov/ons/addressIndex/parsers/postTokenizeTreatmentTest.scala]: should we add another test for input of 

"40  -      45"

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala [hunk 3]
```diff
@@ -113,114 +113,127 @@ class AddressIndexRepository @Inject()(
     * @param tokens tokens for the ES query
     * @return Search definition containing query to the ES
     */
-  def generateQueryAddressRequest(tokens: Seq[CrfTokenResult]): SearchDefinition = {
-
-    val tokensMap = Tokens.tokensToValidMap(tokens)
+  def generateQueryAddressRequest(tokens: Map[String, String]): SearchDefinition = {
 
     val query =
       bool {
```
Comment (iRhysBradbury) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: should we look to factor out all this repeat code into one manageable block?

File: parsers/src/test/scala/uk/gov/ons/addressIndex/parsers/postTokenizeTreatmentTest.scala (hunk 2 above)
Comment (Mironor) [inline parsers/src/test/scala/uk/gov/ons/addressIndex/parsers/postTokenizeTreatmentTest.scala]: At this stage the tokens is "40-45" and that is already covered by the 120-122 case.
If the token is "40 - 45" it's a problem with pre-treatment, not with the post-treatment

File: server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala (hunk 3 above)
Comment (Mironor) [inline server/app/uk/gov/ons/addressIndex/server/modules/AddressIndexRepository.scala]: It is fine like it currently is (explicit).
It is soon to be rewritten completely, so it's pointless to try to refactor it currently.

**Pull request merged**